Feature 1183 memory documentation (#1340)
dtcenter/METplus: docs/Contributors_Guide/add_use_case.rst

  processed for analysis. Users can take an example and modify the run times
  to produce more output as desired.
- No errors should result from running the use case.

> - The amount of required processing memory should not exceed the current limit 

Reviewer: I think this item can be removed. The information about memory usage for running in automation is well documented below with your other additions. This information isn't as much a rule for developing the use case as it is a rule for adding it to the testing environment.

filename in the log output by using the search box on the top right of the
log output.

> If the use case does not run successfully in Github Actions and standard error checking 

Reviewer: Suggest changing this paragraph to something like this:

> Potential reasons for a use case to run successfully in the user's environment but fail in the GitHub Actions environment include:
> * Errors providing input data (see :ref:\`use_case_input_data\`)
> * Using hard-coded paths from the user's machine
> * Referencing variables set in the user's configuration file or local environment
> * Memory usage of the use case exceeds the available memory in the GitHub Actions environment
> 
> Github Actions has \`limited memory \<https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources>`_ available and will cause the use case to fail if exceeded. A failure caused by exceeding the memory allocation in a Python Embedding script may result in an unclear error message. If you suspect that this is the case, consider utilizing a Python memory profiler to check the Python script's memory usage. If your use case exceeds the limit, try to pare down the data held in memory and use less memory intensive Python routines.

new job for the new use case. See the :ref:`subset_category` section and the
multiple medium_range jobs for an example.

.. _exceeded-Github-Actions:

Use Cases That Exceed Memory Allocations of Github Actions
----------------------------------------------------------

If a use case utilizing Python embedding does not run successfully in 
Github Actions due to exceeding the memory limit and memory mitigation 
steps were unsuccessful in lowering memory usage, please take the following steps.

- Document the Github Actions failure in the Github use case issue. 
  Utilize a Python memory profiler to identify as specifically as possible 
  where the script exceeds the memory limit.
- Add the use case to the :ref:`memory-intense-use-cases` list.
- In the internal_tests/use_cases/all_use_cases.txt file, ensure that the 
  use case is listed as the lowest-listed use case in its respective category. 

>   Change the number in front of the new use case to a letter, preceeded 

Reviewer: ```suggestion
  Change the number in front of the new use case to an 'X', preceeded 
```

  by the ‘#’ character::

	#X::GridStat_fcstRTOFS_obsGHRSST_climWOA_sst::model_applications/marine_and_cryosphere/GridStat_fcstRTOFS_obsGHRSST_climWOA_sst.conf, model_applications/marine_and_cryosphere/GridStat_fcstRTOFS_obsGHRSST_climWOA_sst/ci_overrides.conf:: icecover_env, py_embed

> - In the **.github/parm/use_case_groups.json** file, change the “run” variable 

Reviewer: If the use case is in the use_case_groups.json file marked with run = false, then it will still be run on a pull request. Setting run = true will execute the use case in automation for every push event.

A use case that exceeds the memory limit should be excluded from this file. Future enhancements could include adding a "never" value to note that it should never be run in automation but still listed with the rest of the use cases in case we are eventually able to run the case in automation.

I suggest changing this item to remove the item from the JSON file.
